mbuild/formats/lammpsdata.py

```diff
@@ -1159,7 +913,429 @@ def _get_impropers(structure, epsilon_conversion_factor):
+                        type1,
+                        type2,
+                        epsilon,
+                        sigma,
+                        forcefield_dict[type1],
+                        forcefield_dict[type2],
+                    )
+                )
+
+    # Pair coefficients
+    else:
+        if pair_coeff_label:
+            data.write("\nPair Coeffs # {}\n".format(pair_coeff_label))
+        else:
+            data.write("\nPair Coeffs # lj\n")
+
+        if unit_style == "real":
+            data.write("#\tepsilon (kcal/mol)\t\tsigma (Angstrom)\n")
+        elif unit_style == "lj":
+            data.write("#\treduced_epsilon \t\treduced_sigma \n")
+        for idx, epsilon in sorted(epsilon_dict.items()):
+            data.write(
+                "{}\t{:.5f}\t\t{:.5f}\t\t# {}\n".format(
+                    idx, epsilon, sigma_dict[idx], forcefield_dict[idx]
+                )
+            )
+
+    return
+
+
+def _write_bond_information(structure, data, unique_bond_types, unit_style):
+    """Write Bond Coeffs section of lammps data file."""
+    data.write("\nBond Coeffs # harmonic\n")
+    if unit_style == "real":
+        data.write("#\tk(kcal/mol/angstrom^2)\t\treq(angstrom)\n")
+    elif unit_style == "lj":
+        data.write("#\treduced_k\t\treduced_req\n")
+    sorted_bond_types = {
+        k: v
+        for k, v in sorted(unique_bond_types.items(), key=lambda item: item[1])
+    }
+    for params, idx in sorted_bond_types.items():
+        data.write(
+            "{}\t{}\t\t{}\t\t# {}\t{}\n".format(
+                idx,
+                params[0],
+                params[1],
+                params[2][0],
+                params[2][1],
+            )
+        )
+
+
+def _write_angle_information(
+    structure, data, unique_angle_types, use_urey_bradleys, unit_style
+):
+    """Write Angle Coeffs section of lammps data file."""
+    sorted_angle_types = {
+        k: v
+        for k, v in sorted(unique_angle_types.items(), key=lambda item: item[1])
+    }
+    if use_urey_bradleys:
+        data.write("\nAngle Coeffs # charmm\n")
+        data.write(
+            "#\tk(kcal/mol/rad^2)\t\ttheteq(deg)\tk(kcal/mol/angstrom^2)\treq(angstrom)\n"
+        )
+        for params, idx in sorted_angle_types.items():
+            data.write("{}\t{}\t{:.5f}\t{:.5f}\t{:.5f}\n".format(idx, *params))
+
+    else:
+        data.write("\nAngle Coeffs # harmonic\n")
+
+        if unit_style == "lj":
+            data.write("#\treduced_k\t\ttheteq(deg)\n")
+        else:
+            data.write("#\tk(kcal/mol/rad^2)\t\ttheteq(deg)\n")
+
+        for params, idx in sorted_angle_types.items():
+            data.write(
+                "{}\t{}\t\t{:.5f}\t# {}\t{}\t{}\n".format(
+                    idx,
+                    params[0],
+                    params[1],
+                    params[3][0],
+                    params[2],
+                    params[3][1],
+                )
+            )
+
+    return
```

Comment:
I think I understand what you are doing here, but these methods can just not return anything, if I recall correctly